CMakeLists.txt

```diff
@@ -15,6 +15,11 @@ target_include_directories(bls12-381 PUBLIC include)
 set_target_properties(bls12-381 PROPERTIES PUBLIC_HEADER "${BLS12-381_HEADERS}")
 target_compile_features(bls12-381 PUBLIC cxx_std_20)
 
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+  target_sources(bls12-381 PRIVATE src/arithmetic.s)
+  set_source_files_properties(src/arithmetic.s PROPERTIES COMPILE_FLAGS "-Wno-unused-command-line-argument")
```

Comment:
cmake will use the c compiler for assembler files, and also plumb through cflags. clang doesn't like getting passed -O3 etc when invoking the assembler and complains. Don't know a better way to resolve this, we do similar in spring/libraries/chain